module_utils/network_lsr/argument_validator.py

```diff
@@ -1088,8 +1088,10 @@ def _validate_post_state(self, value, name, result):
             actions.append(persistent_state)
 
         # If the profile should be absent at the end, it needs to be present in
-        # the meantime to allow to (de)activate it
-        if persistent_state == "absent" and state:
+        # the meantime to allow to (de)activate it. This is only possible if it
+        # is completely defined, for which `type` needs to be specified.
+        # Otherwise, downing is happening on a best-effort basis
+        if persistent_state == "absent" and state and result.get("type"):
```

Comment:
Does this change the role interface? What happened before the change if persistent_state == "absent" and state is given and type is not present?

Reply:
It depends on whether the profile existed on the system, if it was there, it would be used, otherwise there would be a failure. Type needs to be specified for "present" to work in case the profile is not on the system, otherwise it would just check this.

Reply:
So it is not an interface change, because if the interface had not been there, the role failed, and if it had been there, the `actions.append("present")` was superfluous.

Reply:
yes, with profile instead of interface.